### PR TITLE
Feat/implement player animations zappy gui

### PR DESCRIPTION
**Pull request by mathieurobert1**
I implemented the player's animations:

Now, for each player's state, an animation is displayed.

More details : #111 

image
image

### CODE REVIEW

File: gui/src/GameDatas/Player.cpp [hunk 1]
```diff
@@ -82,6 +101,16 @@ std::string Gui::Player::getBroadcast() const
     return _broadcast;
 }
 
+void Gui::Player::setCurrentFrame(int currentFrame)
+{
+    _currentFrame = currentFrame;
+}
+
+int Gui::Player::getCurrentFrame()
```
Comment (Njord201) [inline gui/src/GameDatas/Player.cpp:109]: Add const label

File: gui/src/GameDatas/Player.cpp [hunk 2]
```diff
@@ -102,3 +131,13 @@ Vector3 Gui::Player::getCenterPosition()
 {
     return {(float)((double)_position.first * SIZE_TILE + SIZE_TILE / 2), 0, (float)((double)_position.second * SIZE_TILE + SIZE_TILE / 2)};
 }
+
+void Gui::Player::restartAnimationTimeEllapsed()
+{
+    _animationTimeEllapsed = clock();
+}
+
+clock_t Gui::Player::getAnimationTimeEllapsed()
```
Comment (Njord201) [inline gui/src/GameDatas/Player.cpp:140]: Add const label

**Review by Njord201 — CHANGES_REQUESTED** [review]
Two little requests, good job!

File: gui/include/GameDatas/Player.hpp [hunk 3]
```diff
@@ -165,6 +179,33 @@ class Gui::Player {
...
+
+        /**
+         * @brief Get the Current Frame object.
+         *
+         * @return int - Current frame.
+         */
+        int getCurrentFrame();
+
+        /**
+         * @brief Restart the timer animation;
```
Comment (theophile-jr) [inline gui/include/GameDatas/Player.hpp:197]: ";"?

File: gui/include/GameDatas/Player.hpp [hunk 4]
```diff
@@ -165,6 +179,33 @@ class Gui::Player {
...
+
+        /**
+         * @brief Get the Current Frame object.
+         *
+         * @return int - Current frame.
+         */
+        int getCurrentFrame() const;
+
+        /**
+         * @brief Restart the timer animation;
```
Comment (Njord201) [inline gui/include/GameDatas/Player.hpp:197]:
```suggestion
         * @brief Restart the timer animation.
```

**Review by Njord201 — CHANGES_REQUESTED** [review]

**Review by theophile-jr — APPROVED** [review]

**Review by Njord201 — APPROVED** [review]

**Review by theophile-jr — APPROVED** [review]
LGTM !  (Sorry for the previous message, missclick)

**Review by Njord201 — APPROVED** [review]

File: gui/include/GameDatas/Team.hpp [hunk 5]
```diff
@@ -185,6 +192,9 @@ class Gui::Team {
 
     private:
 
+
+        ModelAnimation*             _modelAnimation;    // Model to animate players.
```
Comment (Thomaltarix) [inline gui/include/GameDatas/Team.hpp:196]: Use smart pointers, :wink:

File: gui/src/GameDatas/Team.cpp [hunk 6]
```diff
@@ -87,6 +89,11 @@ Model Gui::Team::getPlayerModel() const
     return _playerModel;
 }
 
+ModelAnimation *Gui::Team::getPlayerModelAnimation() const
```
Comment (Thomaltarix) [inline gui/src/GameDatas/Team.cpp:92]: Same here, use a smart pointer

**Review by Thomaltarix — APPROVED** [review]
Unless the raw pointer, great job, it's great

**Review by Thomaltarix — CHANGES_REQUESTED** [review]
You just have a raw pointer

**Review by Thomaltarix — APPROVED** [review]
My bad, i didn't know that it was raylib diff

**Pull request merged**